### DISCUSSION TITLE
Question on OPC UA

### DISCUSSION THREAD

**Discussion by RakshanPremsagar** [eclipse-milo/milo#755] [General]
Hello Kevin,
I have a query, or rather an observation which I want to make sure if my understanding is right.

Case1: I have 2 hosts(HOSTA and HOSTB), both have LDS servers with multicast enabled. I run the HOSTA ( in this perspective is a client) LDS ME server, next I run the HOSTB LDSME Server. On the host A log, I see that a new server is detected( from HOST B). Next I run another server(eg: a basic OPC UA server) on HOST B and register it to the LDSME of HOST B.
I observe the log of HOST A, which says that the new server is also detected.

So if i run the findServersOnNetwork from the HOSTA, I get all three servers.

Case2: Again 2 hosts (HOSTA and HOSTB)both have LDS servers with multicast enabled. I run the HOSTB LDSME Server. Next, I run another server (eg: a basic OPC UA server) on HOST B and register it to the LDSME of HOST B. Now I run the I run the HOSTA ( in this perspective is a client) LDS ME server. The log of HOST A only finds the basic OPC UA server which was registered to the LDSME. It does not recognise the LDS ME of the HOSTB.

So if i run the findServersOnNetwork from the HOSTA, I get two servers.(LDS ME of HOSTA and the registered server. LDSME of HOST B is not seen)

Reasoning: In the case1, since the LDS ME(HOSTB) runs well before another server registers itself, the records of LDS ME is announced. The LDS ME of the HOSTA, will receive these announcements and detect this server. Next when the basic OPC UA server registers itself to LDSME of HOSTB, again the new server announcements will be picked up by the HOSTA LDSME.
Thus, 3 servers are seen while running FindServersOnNetwork.

In the case2, although LDSME (HOSTB) announces itself, the basic server registers to the LDS ME and then the announcements of the new registered server will be present in the network. Thus, when the HOSTA LDSME runs next, it will receive the announcements of the registered server and not the LDSME of the HOSTB, thus giving us 2 servers while running FindServersOnNetwork.

Please let me know if my understanding is right? Feel free to correct me if I am wrong here.

Thanks,
Rakshan

**Comment by kevinherron** [comment]
Sorry, I'm not sure whether you're right or wrong. Your explanation sounds reasonable but I'm not an expert on the discovery spec.

**Reply by RakshanPremsagar** [reply]
Oh ok. But thanks for your reply :)
